CoreFoundation/String.subproj/CFString.c

```diff
@@ -3233,7 +3233,7 @@ Boolean CFStringFindWithOptionsAndLocale(CFStringRef string, CFStringRef stringT
 
                                     do {
                                         str1Char = CFStringGetCharacterFromInlineBuffer(&inlineBuf1, --index);
-                                    } while (CFUniCharIsMemberOfBitmap(str1Char, graphemeBMP), (rangeToSearch.location < index));
+                                    } while (CFUniCharIsMemberOfBitmap(str1Char, graphemeBMP) && (rangeToSearch.location < index));
```

Comment:
oh, sorry, might be wrong. If `CFUniCharIsMemberOfBitmap(str1Char, graphemeBMP)` doesn't have any side-effects, the original code looks wrong.